src/main/java/org/jenkins/tools/test/gradle/ExternalGradleRunner.java

```diff
@@ -0,0 +1,148 @@
+package org.jenkins.tools.test.gradle;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.lang.SystemUtils;
+import org.jenkins.tools.test.exception.GradleExecutionException;
+import org.jenkins.tools.test.model.PluginCompatTesterConfig;
+
+public class ExternalGradleRunner implements GradleRunner {
+
+    private static final Logger LOGGER = Logger.getLogger(ExternalGradleRunner.class.getName());
+
+    @CheckForNull
+    private final File externalGradle;
+
+    @CheckForNull
+    private final File gradleProperties;
+
+    @NonNull
+    private final List<String> gradleArgs;
+
+    public ExternalGradleRunner(
+            @CheckForNull File externalGradle, @CheckForNull File gradleProperties, @Nullable List<String> gradleArgs) {
+        this.externalGradle = externalGradle;
+        this.gradleProperties = gradleProperties;
+        this.gradleArgs = gradleArgs == null ? List.of() : List.copyOf(gradleArgs);
+    }
+
+    public ExternalGradleRunner(@NonNull PluginCompatTesterConfig config) {
+        this(config.getExternalGradle(), config.getGradleProperties(), config.getGradleArgs());
+    }
+
+    @Override
+    @SuppressFBWarnings(value = "COMMAND_INJECTION", justification = "intended behavior")
+    public void run(
+            Map<String, String> properties,
+            File baseDirectory,
+            @CheckForNull String moduleName,
+            @CheckForNull File buildLogFile,
+            String... tasks)
+            throws GradleExecutionException {
+
+        List<String> cmd = new ArrayList<>();
+
+        File gradleWrapper = new File(baseDirectory, SystemUtils.IS_OS_WINDOWS ? "gradlew.bat" : "gradlew");
+        if (externalGradle != null) {
+            cmd.add(externalGradle.getAbsolutePath());
+        } else if (gradleWrapper.exists() && gradleWrapper.canExecute()) {
+            cmd.add(gradleWrapper.getAbsolutePath());
+        } else {
+            cmd.add("gradle");
+        }
+
+        cmd.add("--no-daemon");
```

Comment:
Why? Daemon mode is much better performance-wise, and it is the default one these days